src/ContainerConfig.php

```diff
@@ -85,12 +91,30 @@ public function define(LaravelIocContainer $container)
 
         $container->bindValue(Container::class, $container->getLaravelContainer());
 
-        $container->bindCallback(IIocContainer::SCOPE_SINGLETON, Session::class, function () {
-            $session_factory = new SessionFactory;
-            $session = $session_factory->newInstance($_COOKIE);
-            return $session;
```

Comment:
ServerRequestInterface is immutable, no value is expected / should change. This is not Client Request.